libsycl/CMakeLists.txt

```diff
@@ -117,10 +115,23 @@ add_custom_command(
 install(DIRECTORY "${LIBSYCL_SOURCE_INCLUDE_DIR}/sycl" DESTINATION ${LIBSYCL_INCLUDE_DIR} COMPONENT sycl-headers)
 install(DIRECTORY "${LIBSYCL_SOURCE_INCLUDE_DIR}/CL" DESTINATION ${LIBSYCL_INCLUDE_DIR} COMPONENT sycl-headers)
 
-set(LIBSYCL_RT_LIBS ${LIBSYCL_SHARED_OUTPUT_NAME})
-
-add_subdirectory(src)
+set(LIBSYCL_LIB_NAME "sycl")
+set(LIBSYCL_SHARED_OUTPUT_NAME "${LIBSYCL_LIB_NAME}")
+if (CMAKE_SYSTEM_NAME STREQUAL Windows)
+  if (CMAKE_MSVC_RUNTIME_LIBRARY AND (NOT CMAKE_MSVC_RUNTIME_LIBRARY MATCHES "DLL$"))
+      message(FATAL_ERROR "libsycl requires a DLL version of the MSVC CRT.")
+  endif()
+  if ((NOT CMAKE_MSVC_RUNTIME_LIBRARY AND uppercase_CMAKE_BUILD_TYPE STREQUAL "DEBUG")
+      OR (CMAKE_MSVC_RUNTIME_LIBRARY STREQUAL "MultiThreadedDebugDLL"))
+    set(LIBSYCL_SHARED_OUTPUT_NAME "${LIBSYCL_SHARED_OUTPUT_NAME}d")
+  endif()
```

Comment:
`CMAKE_BUILD_TYPE` is only relevant for single-configuration CMake generators, so this won't have an impact for multi-configuration (IDE) generators like for msbuild. For multi-configuration generators, I think the result will be that release and debug builds will use the same name (but be placed in separate directories).

I also remain skeptical that changing the name of the DLL based on build configuration is desirable. I do see that there is some existing precedent for OpenMP though.

Also, `uppercase_CMAKE_BUILD_TYPE` doesn't seem to be assigned in this file. Is it being supplied from elsewhere? Perhaps this should use `TOUPPER "${CMAKE_BUILD_TYPE}"`.